mailbox/api/src/main/java/org/apache/james/mailbox/model/Mailbox.java

```diff
@@ -19,78 +19,158 @@
 package org.apache.james.mailbox.model;
 
 import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MailboxUtil;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
 
 /**
  * Models long term mailbox data.
  */
-public interface Mailbox {
-    void setMailboxId(MailboxId id);
-    
+public class Mailbox {
+    private MailboxId id = null;
```

Comment:
should it be Optional?

Reply:
Nope, I did not intend to refactor this class